cmd/apiProxyList_test.go

```diff
@@ -0,0 +1,68 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/apim"
+	apimhttp "github.com/SAP/jenkins-library/pkg/apim"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRunApiProxyList(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Get API Proxy List successfull test", func(t *testing.T) {
+		config := getDefaultOptionsForApiProxyList()
+		httpClientMock := &apimhttp.HttpMockAPIM{StatusCode: 200, ResponseBody: `{"some": "test"}`}
+		seOut := apiProxyListCommonPipelineEnvironment{}
+		apim := apim.Bundle{APIServiceKey: config.APIServiceKey, Client: httpClientMock}
+		// test
+		err := getApiProxyList(&config, apim, &seOut)
+
+		// assert
+		if assert.NoError(t, err) {
+			assert.EqualValues(t, seOut.custom.APIProxyList, "{\"some\": \"test\"}")
+			t.Run("check url", func(t *testing.T) {
+				assert.Equal(t, "/apiportal/api/1.0/Management.svc/APIProxies?filter=isCopy+eq+false&$orderby=name&$skip=1&$top=4", httpClientMock.URL)
+			})
+			t.Run("check method", func(t *testing.T) {
+				assert.Equal(t, "GET", httpClientMock.Method)
+			})
+		}
+	})
+
+	t.Run("Get API Proxy List failed test", func(t *testing.T) {
+		config := getDefaultOptionsForApiProxyList()
+		httpClientMock := &apimhttp.HttpMockAPIM{StatusCode: 400}
+		seOut := apiProxyListCommonPipelineEnvironment{}
+		apim := apim.Bundle{APIServiceKey: config.APIServiceKey, Client: httpClientMock}
+		// test
+		err := getApiProxyList(&config, apim, &seOut)
+		// assert
+		assert.EqualError(t, err, "HTTP GET request to /apiportal/api/1.0/Management.svc/APIProxies?filter=isCopy+eq+false&$orderby=name&$skip=1&$top=4 failed with error: : Bad Request")
+	})
+
+	t.Run("Get API Proxy List failed test for empty odata filters", func(t *testing.T) {
+		config := apiProxyListOptions{
+			APIServiceKey: apimhttp.GetServiceKey(),
+		}
+		httpClientMock := &apimhttp.HttpMockAPIM{StatusCode: 400}
```

Comment:
This is not a different test to the one above, as you are mocking the response code. How about checking for an empty query and failing here in Piper, instead of sending known faulty requests to the backend?

Reply:
empty query doesn't fail the test, as all odata filters are optional in nature.